### ISSUE TITLE
Create a way to imbed youtube videos to play on the videos and demonstrations page. 

### ISSUE THREAD

**Issue by ashleyhorner** [mieweb/docs#54]
**Is your feature request related to a problem? Please describe.**
When we publish videos to the google drive and then add them to the help docs using an ID, the video is sometimes blocked. This error causes people to lose access to the video. It would be better if we could host the videos on youtube but still have the ability to watch them straight on the docs.enterprisehealth.com website. 

**Describe the solution you'd like**
When creating documentation, I need a way to link a youtube video on the google doc page so that when it is commited, the video will display a thumbnail and play directly on the site. 

**Describe alternatives you've considered**
The alternative is guaranteeing that google drive does not block certain videos from being viewed. (See screenshot attached)

**Additional context**
<img width="1512" alt="Screenshot 2024-10-24 at 3 23 35 PM" src="https://github.com/user-attachments/assets/025edbd0-83af-467b-b808-449a3dd89998">

**Comment by horner** [comment]
@ggodlewski didn't we do this already?  I can't find it in the code.

**Comment by horner** [comment]
<img width="831" alt="image" src="https://github.com/user-attachments/assets/df4fcef9-1fdd-476b-917e-8141c5a96e93">

**Comment by monpie3** [comment]
We can create another shortcode, similar to `gvideo`, but specifically for YouTube embeds. 
https://github.com/mieweb/docs/blob/master/themes/wc-eh-docs/layouts/shortcodes/gvideo.html 

Additionally, we could add additional attributes like time and title for more flexibility:  https://www.kiroule.com/article/use-video-embeds-in-hugo-theme/

**Comment by monpie3** [comment]
I'll try to republish Videos and demonstrations page using built-in hugo shortcode:
`{{< youtube __R9twSkS2A >}}`

**Comment by monpie3** [comment]
Okay, this worked:
https://docs.enterprisehealth.com/videos-and-demonstrations/#generative-ai

It's larger than the rest of the videos, but I think we can easily fix this by adding `youtube.html` in [/themes/wc-eh-docs/layouts/shortcodes](https://github.com/mieweb/docs/blob/master/themes/wc-eh-docs/layouts/shortcodes/) with the appropriate html 

So, this can be a way to format a YouTube video in Google Docs:
`{{< youtube ID_FROM_YOUTUBE_VIDEO_URL>}}` 


When the YouTube link looks like: youtube.com/watch?v=**__R9twSkS2A** , it will be: `{{< youtube __R9twSkS2A >}}`
                                              
 ----

I guess there’s no point in expanding this tag further, so I would just hardcode the size and width in HTML. What do you think @ggodlewski  and @horner ?

**Comment by ggodlewski** [comment]
> @ggodlewski didn't we do this already? I can't find it in the code.

@horner we've done this for youtube videos. You can transform yt links to player. It's just not configured.

Example: https://dev.wikigdrive.com/drive/0AF_nrE0_QH_2Uk9PVA#drive_config

![image](https://github.com/user-attachments/assets/78ab1f7c-e4a4-4ecb-9b23-4f23ecde53d7)

This converts yt links into @monpie3 code:

https://dev.wikigdrive.com/drive/0AF_nrE0_QH_2Uk9PVA/content/functions/quality-of-care/create-file-for-submission.md#markdown

![image](https://github.com/user-attachments/assets/1b6f8c74-15a3-40e3-9f71-a8080d1c72c1)

If that is ok, all you need is to copy this rule form prod to dev.

I think we also discuss with @wreiske about uploading google videos to cloudflare. That would require writing additional code (cloudflare worker, or wgd upload and rewriting links).

**Comment by monpie3** [comment]
In our last meeting, we discussed that, since we also have YouTube videos linked elsewhere in the documentation, (for example: https://docs.enterprisehealth.com/functions/quality-of-care/create-file-for-submission/), it would be a good idea to use a separate YouTube tag. This approach will also ensure consistency with the `gvideo` tag.
The youtube tag can be used as:
`{{< youtube __R9twSkS2A >}}` or `{{< youtube id="__R9twSkS2A" >}}`

I added a PR that resolves the current issue where embedded YouTube videos have a different size than embedded Google Drive videos. It is waiting for code review.
